provisioning_client/tests/hsm_client_tpm_ut/hsm_client_tpm_ut.c

```diff
@@ -127,6 +127,72 @@ static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
 
 static TEST_MUTEX_HANDLE g_testByTest;
 
+char* umock_stringify_TPMA_SESSION(const TPMA_SESSION* value)
+{
+    char* result = (char*)my_gballoc_malloc(1);
+    (void)value;
+    result[0] = '\0';
+    return result;
+}
+
+int umock_are_equal_TPMA_SESSION(const TPMA_SESSION* left, const TPMA_SESSION* right)
+{
+    int result;
+
+    if (((left == NULL) && (right != NULL)) || ((right == NULL) && (left != NULL)))
+    {
+        result = 0;
+    }
+    else
+    {
+        if (left == right)
+        {
+            result = 1;
+        }
+        else
+        {
+            result = 0;
+        }
+    }
+
+    return result;
+}
+
+int umock_copy_TPMA_SESSION(TPMA_SESSION* destination, const TPMA_SESSION* source)
+{
+    int result;
+
+    if ((source == NULL) || (destination == NULL))
+    {
+        result = -1;
+    }
+    else
+    {
+        destination->continueSession = source->continueSession;
+        destination->auditExclusive = source->auditExclusive;
+        destination->auditReset = source->auditReset;
+        destination->Reserved_at_bit_3 = source->Reserved_at_bit_3;
+        destination->decrypt = source->decrypt;
+        destination->encrypt = source->encrypt;
+        destination->audit = source->audit;
+        result = 0;
+    }
+
+    return result;
+}
+
+void umock_free_TPMA_SESSION(TPMA_SESSION* value)
+{
+    // reset values
+    value->continueSession = 0;
```

Comment:
This should not be needed at all. Only if memory is allocated.

Reply:
removed all resets